src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs

```diff
@@ -29,32 +22,47 @@ public string ReadConfigurationXml()
             // Config file should be in the base directory of the app domain
             string configFilePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "ApplicationInsights.config");
 
-            // Ensure config file actually exists
-            if (File.Exists(configFilePath))
+            try
             {
-                return File.ReadAllText(configFilePath);
+                // Ensure config file actually exists
+                if (File.Exists(configFilePath))
+                {
+                    return File.ReadAllText(configFilePath);
+                }
+            }
+            catch (FileNotFoundException)
+            {
+                // For cases when file was deleted/modified while reading
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
+            }
+            catch (DirectoryNotFoundException)
+            {
+                // For cases when file was deleted/modified while reading
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
+            }
+            catch (IOException)
+            {
+                // For cases when file was deleted/modified while reading
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
+            }
+            catch (UnauthorizedAccessException)
+            {
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
+            }
+            catch (SecurityException)
+            {
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
```

Comment:
PathTooLong is IOException. NotSupportedException is thrown when path is in incorrect format - we create path on line 23; it is not possible to have incorrect format.

---
In reply to: [77935073](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/304#discussion_r77935073) [](ancestors = 77935073)